mlflow/crewai/autolog.py

```diff
@@ -0,0 +1,228 @@
+    config = AutoLoggingConfig.init(flavor_name=mlflow.gemini.FLAVOR_NAME)
+
+    if config.log_traces:
+        fullname = f"{self.__class__.__name__}.{original.__name__}"
+        with mlflow.start_span(
+            name=fullname,
+            span_type=_get_span_type(self.__class__, fullname),
+        ) as span:
+            inputs = _construct_full_inputs(original, self, *args, **kwargs)
+            span.set_inputs(inputs)
+            _set_span_attributes(span=span, instance=self)
+            result = original(self, *args, **kwargs)
+            # need to convert the response of generate_content for better visualization
+            outputs = result.__dict__ if hasattr(result, "__dict__") else result
+            span.set_outputs(outputs)
+
+            return result
+
+
+def _get_span_type(cls, task_name: str) -> str:
+    from crewai.flow.flow import Flow
+
+    if issubclass(cls, Flow):
+        return SpanType.CHAIN
+    span_type_mapping = {
+        "Crew.kickoff": SpanType.CHAIN,
+        "Crew.kickoff_for_each": SpanType.CHAIN,
+        "Crew.train": SpanType.CHAIN,
+        "Agent.execute_task": SpanType.AGENT,
+        "Task.execute_sync": SpanType.CHAIN,
+        "ShortTermMemory.save": SpanType.RETRIEVER,
+        "ShortTermMemory.search": SpanType.RETRIEVER,
+        "ShortTermMemory.reset": SpanType.RETRIEVER,
+        "LongTermMemory.save": SpanType.RETRIEVER,
+        "LongTermMemory.search": SpanType.RETRIEVER,
+        "LongTermMemory.reset": SpanType.RETRIEVER,
+        "UserMemory.search": SpanType.RETRIEVER,
+        "UserMemory.save": SpanType.RETRIEVER,
+        "EntityMemory.search": SpanType.RETRIEVER,
+        "EntityMemory.save": SpanType.RETRIEVER,
+        "EntityMemory.reset": SpanType.RETRIEVER,
+        "Knowledge.query": SpanType.RETRIEVER,
+        "LLM.call": SpanType.LLM,
+        "Flow.kickoff": SpanType.CHAIN,
+    }
+    return span_type_mapping.get(task_name, SpanType.UNKNOWN)
+
+
+def _is_serializable(value):
+    try:
+        json.dumps(value, cls=TraceJSONEncoder, ensure_ascii=False)
+        return True
+    except (TypeError, ValueError):
+        return False
+
+
+def _construct_full_inputs(func, *args, **kwargs):
+    signature = inspect.signature(func)
+    # this does not create copy. So values should not be mutated directly
+    arguments = signature.bind_partial(*args, **kwargs).arguments
+
+    if "self" in arguments:
+        arguments.pop("self")
+
+    # Avoid non serializable objects and circular references
+    return {
+        k: v.__dict__ if hasattr(v, "__dict__") else v
+        for k, v in arguments.items()
+        if v is not None and _is_serializable(v)
+    }
+
+
+def _set_span_attributes(span: LiveSpan, instance):
+    instance_name = instance.__class__.__name__
+    ## memory does not have helpful attributes
+    if instance_name == "Crew":
+        for key, value in instance.__dict__.items():
+            key = f"crewai.crew.{key}"
```

Comment:
We don't use any prefix for attributes in other flavors. Can we remove them for consistency?